orttraining/orttraining/models/runner/training_runner.cc

```diff
@@ -92,7 +108,7 @@ Status TrainingRunner::Initialize() {
   }
 
   // always configure the loss function
-  {
+  if (params_.num_pipeline_stages == 1 || params_.mpi_context.world_rank == params_.mpi_context.world_size - 1) {
```

Comment:
>params_.mpi_context.world_rank == params_.mpi_context.world_size - 1 [](start = 42, length = 68)

what about D+H+P?

Reply:
This change is correct to them. Whether loss presents only depends on pipeline partition.